### PR TITLE
fix edge dragging and method argument assignment

### PR DESCRIPTION
**Pull request by Frizi**
### Pull Request Description

Fixed broken edge dragging and creating nodes from ports. Added basic support for multiple output ports, driven by already existing analysis of the port binding structure. Those constructs are not yet supported by the engine (hence the error in code), but the IDE has easier time already dealing with ports as individual binding expressions, not whole nodes.

image

Also improved the ability to interpret applied method arguments. Different cases of dynamic, static calls or partially applied functions are now properly supported.

image

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui2/src/providers/functionInfo.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { identity } from '@vueuse/core'
+import type { ExprId } from 'shared/yjsModel'
+import { createContextStore } from '.'
+
+interface FunctionInfo {
+  callId: ExprId | undefined
+}
+
+export { injectFn as injectFunctionInfo, provideFn as provideFunctionInfo }
+const { provideFn, injectFn } = createContextStore('Port info', identity<FunctionInfo>)
```
Comment (farmaazon) [inline app/gui2/src/providers/functionInfo.ts]: I feel no information can be derived from module name nor its path that this is "Enso" function used for widgets. Only from the name used in `createContextStore` but it seems like overlook.

File: app/gui2/src/providers/widgetUsageInfo.ts [hunk 2]
```diff
@@ -11,8 +11,18 @@ const { provideFn, injectFn } = createContextStore('Widget usage info', identity
  * AST node.
  */
 interface WidgetUsageInfo {
-  input: WidgetInput
+  usageKey: unknown
```
Comment (farmaazon) [inline app/gui2/src/providers/widgetUsageInfo.ts:21]: I think a short explanation how this key is used would be helpful.

File: app/gui2/src/stores/graph/graphDatabase.ts [hunk 3]
```diff
@@ -204,28 +204,43 @@ export class GraphDb {
...
 
   identifierUsed(ident: string): boolean {
     return this.bindings.identifierToBindingId.hasKey(ident)
   }
 
-  isMethodCall(id: ExprId): boolean {
-    return this.getExpressionInfo(id)?.methodCall != null
+  isKnownFunction(id: ExprId): boolean {
+    return this.getMethodCallInfo(id) != null
+  }
```
Comment (farmaazon) [inline app/gui2/src/stores/graph/graphDatabase.ts]: I would rename to `isKnownFunctionCall`.

**Review by MichaelMauderer — APPROVED** [review]

File: app/gui2/src/util/callTree.ts [hunk 4]
```diff
@@ -156,13 +215,37 @@ const unknownArgInfoNamed = (name: string) => ({
...
+}
+
+export class ForcePort {
+  constructor(public ast: AstExtended) {
+    if (ast instanceof ForcePort) throw new Error('ForcePort cannot be nested')
+  }
+  [GetUsageKey]() {
+    return this.ast
+  }
+}
```
Comment (farmaazon) [inline app/gui2/src/util/callTree.ts]: Why it's here? Seems a bit unrelated to rest of the code. And docs would be helpful.

**Review by farmaazon — APPROVED** [review]
